org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/wizard/welcome/GradleWelcomeWizardPage.java

```diff
@@ -71,62 +83,68 @@ protected void createWidgets(Composite root) {
         showWelcomePageCheckboxLayoutData.widthHint = welcomeTextLayoutData.widthHint;
         showWelcomePageCheckboxLayoutData.verticalIndent = 15;
         showWelcomePageCheckbox.setLayoutData(showWelcomePageCheckboxLayoutData);
-        showWelcomePageCheckbox.setSelection(((ProjectImportWizard)getWizard()).isShowWelcomePage());
+        showWelcomePageCheckbox.setSelection(((AbstractProjectWizard) getWizard()).isShowWelcomePage());
         showWelcomePageCheckbox.addSelectionListener(new SelectionAdapter() {
 
             @Override
-            public void widgetSelected(SelectionEvent e) {
-                ((ProjectImportWizard) getWizard()).setWelcomePageEnabled(showWelcomePageCheckbox.getSelection());
+            public void widgetSelected(SelectionEvent event) {
+                @SuppressWarnings("deprecation")
+                ConfigurationScope configurationScope = new ConfigurationScope();
+                IEclipsePreferences node = configurationScope.getNode(UiPlugin.PLUGIN_ID);
+                node.putBoolean(
+                        GradleWelcomeWizardPage.this.welcomePageConfigurator.getWelcomePageEnabledPreferenceKey(),
+                        showWelcomePageCheckbox.getSelection());
+                try {
+                    node.flush();
+                } catch (BackingStoreException e) {
+                    throw new GradlePluginsRuntimeException(e);
+                }
             }
```

Comment:
The logic of reading/writing the preference is not spread across two classes (GradleWelcomeWizardPage and Wizard). This is a step back, in my opinion.